bin/datatool/src/util.rs

```diff
@@ -0,0 +1,417 @@
+};
+
+/// Sequencer key environment variable.
+const SEQKEY_ENVVAR: &str = "STRATA_SEQ_KEY";
+
+/// Operator key environment variable.
+const OPKEY_ENVVAR: &str = "STRATA_OP_KEY";
+
+/// The default network to use.
+///
+/// Right now this is [`Network::Signet`].
+const DEFAULT_NETWORK: Network = Network::Signet;
+
+/// Resolves a [`Network`] from a string.
+pub(super) fn resolve_network(arg: Option<&str>) -> anyhow::Result<Network> {
+    match arg {
+        Some("signet") => Ok(Network::Signet),
+        Some("regtest") => Ok(Network::Regtest),
+        Some(n) => anyhow::bail!("unsupported network option: {n}"),
+        None => Ok(DEFAULT_NETWORK),
+    }
+}
+
+/// Executes a `gen*` subcommand.
+pub(super) fn exec_subc(cmd: Subcommand, ctx: &mut CmdContext) -> anyhow::Result<()> {
+    match cmd {
+        Subcommand::Xpriv(subc) => exec_genxpriv(subc, ctx),
+        Subcommand::SeqPubkey(subc) => exec_genseqpubkey(subc, ctx),
+        Subcommand::SeqPrivkey(subc) => exec_genseqprivkey(subc, ctx),
+        Subcommand::OpXpub(subc) => exec_genopxpub(subc, ctx),
+        Subcommand::Params(subc) => exec_genparams(subc, ctx),
+    }
+}
+
+/// Executes the `genxpriv` subcommand.
+///
+/// Generates a new [`Xpriv`] that will [`Zeroize`](zeroize) on [`Drop`] and writes it to a file.
+fn exec_genxpriv(cmd: SubcXpriv, ctx: &mut CmdContext) -> anyhow::Result<()> {
+    if cmd.path.exists() && !cmd.force {
+        anyhow::bail!("not overwriting file, add --force to overwrite");
+    }
+
+    let xpriv = gen_priv(&mut ctx.rng, ctx.bitcoin_network);
+    let buf = xpriv.encode();
+    let s = base58::encode_check(&buf);
+    fs::write(&cmd.path, s.as_bytes())?;
+
+    Ok(())
+}
+
+/// Executes the `genseqpubkey` subcommand.
+///
+/// Generates the sequencer [`Xpub`] from the provided [`Xpriv`]
+/// and prints it to stdout.
+fn exec_genseqpubkey(cmd: SubcSeqPubkey, _ctx: &mut CmdContext) -> anyhow::Result<()> {
+    let Some(xpriv) = resolve_xpriv(&cmd.key_file, cmd.key_from_env, SEQKEY_ENVVAR)? else {
+        anyhow::bail!("privkey unset");
+    };
+
+    let seq_keys = SequencerKeys::new(&xpriv)?;
+    let seq_xpub = seq_keys.derived_xpub();
+    let raw_buf = seq_xpub.to_x_only_pub().serialize();
+    let s = base58::encode_check(&raw_buf);
+
+    println!("{s}");
+
+    Ok(())
+}
+
+/// Executes the `genseqprivkey` subcommand.
+///
+/// Generates the sequencer [`Xpriv`] that will [`Zeroize`](zeroize) on [`Drop`] and prints it to
+/// stdout.
+fn exec_genseqprivkey(cmd: SubcSeqPrivkey, _ctx: &mut CmdContext) -> anyhow::Result<()> {
+    let Some(xpriv) = resolve_xpriv(&cmd.key_file, cmd.key_from_env, SEQKEY_ENVVAR)? else {
+        anyhow::bail!("privkey unset");
+    };
+
+    let seq_keys = SequencerKeys::new(&xpriv)?;
+    let seq_xpriv = seq_keys.derived_xpriv();
+    let raw_buf = seq_xpriv.to_priv().to_bytes();
+    let s = base58::encode_check(&raw_buf);
```

Comment:
These are secret values that are not zeroized.

Reply:
Done in 965aaee1b7b4318eb8a72b0a873ab5592b15bde6

Reply:
Yes, check 900251ca4eebb44ec80186000302f51762c2684a